packaging/manager.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package packaging
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/packaging/config"
+	"github.com/juju/packaging/manager"
+)
+
+var logger = loggo.GetLogger("juju.packaging")
+
+// PackageManagerName describes a package manager.
+type PackageManagerName string
+
+// The list of supported package managers.
+const (
+	AptPackageManager    PackageManagerName = "apt"
+	YumPackageManager                       = "yum"
```

Comment:
Do they all need PackageManagerName, aren't the later consts just 'string' types?

Reply:
I prefer typed strings as a hint to anyone working with the dependency code that they need to use one of the pre-defined package manager names instead of returning a plain string which could cause issues if misspelled or if we decide to change the package manager name (e.g. apt -> apt-get).

If you think this is overkill, I don't mind changing it to use strings instead.

Reply:
I actually agree, my point was YumPackageManager is just a 'string' type, isn't it?
https://play.golang.org/p/G1pQnXnO22q

Reply:
Doh!!! I somehow thought that they would inherit the type like when using iota...
But obviously that's not the case in general. Nice catch! 

I would have expected the compiler to catch this when assigning to the Package field instead of auto-casting because under the hood it's the same type: https://play.golang.org/p/punXuLV3eAC